src/modules/SdnDiag.LoadBalancerMux/private/Connect-SlbManager.ps1

```diff
@@ -0,0 +1,48 @@
+function Connect-SlbManager {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $false)]
+        [System.String]$SlbManagerPrimary = $env:COMPUTERNAME
+    )
+
+    if (Test-ComputerNameIsLocal -ComputerName $SlbManagerPrimary) {
+        $useLoopback = $true
+    }
+
+    $slbClient = Get-SlbClient -ErrorAction Stop
+    if ($null -ieq $slbClient) {
+        throw "Unable to connect to SlbClient"
+    }
+
+    # if we have already detected that we are using the loopback address, then we can just use that
+    # otherwise we will test to check if the SlbManagerPrimary is an IP address or a hostname
+    # if it is a hostname, then we will resolve it to an IP address
+    if ($useLoopback) {
+        $ipAddress = [System.Net.IPAddress]::Loopback
+    }
+    else {
+        $isIpAddress = ($SlbManagerPrimary -as [IPAddress]) -as [Bool]
```

Comment:
Currently the callers need to have code detect SlbManagerPrimary with same code. Is it proper to move SlbManagerPrimary detection code in Connect-SlbManager so that remove the parameters requirement to make it easy for any caller.

Reply:
@luyaof makes sense. I will move the ability to determine primary into the Connect-SlbManager code.